cli/src/spend_utils.rs

```diff
@@ -102,15 +111,40 @@ where
         )?;
         Ok((message, spend))
     } else {
-        let from_balance = rpc_client
-            .get_balance_with_commitment(from_pubkey, commitment)?
-            .value;
-        let from_rent_exempt_minimum = if amount == SpendAmount::RentExempt {
-            let data = rpc_client.get_account_data(from_pubkey)?;
-            rpc_client.get_minimum_balance_for_rent_exemption(data.len())?
-        } else {
-            0
-        };
+        let account = rpc_client
+            .get_account_with_commitment(from_pubkey, commitment)?
+            .value
+            .unwrap_or_default();
+        let mut from_balance = account.lamports;
+        let from_rent_exempt_minimum =
+            if amount == SpendAmount::RentExempt || amount == SpendAmount::Available {
+                rpc_client.get_minimum_balance_for_rent_exemption(account.data.len())?
+            } else {
+                0
+            };
+        if amount == SpendAmount::Available && account.owner == solana_sdk::stake::program::id() {
+            let state = stake::get_account_stake_state(
+                rpc_client,
+                from_pubkey,
+                account,
+                true,
+                None,
+                false,
+                None,
+            )?;
+            let mut sub_rent_exempt = false;
```

Comment:
nit: I got so confused about this variable name, since I thought `sub` meant that the account was somehow below rent-exemption. `subtract_rent_exempt_minimum` maybe?